interpreter/cling/lib/Interpreter/PerfJITEventListener.cpp

```diff
@@ -0,0 +1,129 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author: Guilherme Amadio <amadio@cern.ch>
```

Comment:
Where does this come from / what is it based on? It might be worth pointing out (in a comment) the source even if it's not a straight / partial copy, to tell us what to look at for future llvm upgrades.

Reply:
I looked at using `interpreter/llvm/src/lib/ExecutionEngine/PerfJITEvents/PerfJITEventListener.cpp` before I implemented this, as the JIT listener from LLVM did not work for me (it uses perf's [jitdump specification](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/perf/Documentation/jitdump-specification.txt) rather than perf map files). I don't expect this file to need many changes in the future. I can add a comment, but the important thing is to just follow `llvm::JITEventListener`'s interfaces, as we derive from that class, so it's probably fine to leave as is.

Reply:
Also, there's already a [comment](https://github.com/root-project/root/pull/11882/files#diff-374f5def4dc7e413f703bac10548ff13d01bb20dd86c730ee5e57d881862c92eR69-R73) further down pointing to the LLVM listener...